static/src/javascripts/projects/common/modules/experiments/tests/newsletter-embed-test.js

```diff
@@ -38,22 +48,20 @@ export const newsletterEmbeds: ABTest = {
             id: 'variant',
             test: (): void => {
                 const iframes = ((document.querySelectorAll('.email-sub__iframe'): NodeList<any>): NodeList<HTMLIFrameElement>);
-                iframes.forEach( (ifrm: HTMLIFrameElement) => {
-                    if (ifrm.id !== 'footer__email-form') {
-                        const doc = ifrm.contentDocument ? ifrm.contentDocument : ifrm.contentWindow.document;
-                        window.addEventListener('load', () => {
+                if (iframes) {
+                    iframes.forEach((ifrm: HTMLIFrameElement) => {
+                        if (ifrm.id !== 'footer__email-form') {
+                            const doc = ifrm.contentDocument ? ifrm.contentDocument : ifrm.contentWindow.document;
                             if (doc) {
-                                const oldDesign = doc.querySelector('.js-ab-embed-old-design');
-                                const newDesign = doc.querySelector('.js-ab-embed-new-design');
-                                if (oldDesign && newDesign) {
-                                    oldDesign.classList.add("hide-element");
-                                    newDesign.classList.remove("hide-element");
-                                    trackComponentInOphan(ifrm.id, 'variant');
+                                if (doc.readyState !== 'complete') {
+                                    window.addEventListener('load', () => triggerVariant(doc, ifrm.id));
```

Comment:
I also tried with `ifrm.addEventListener` here but the event wouldn't always be triggered!